charon/storage.py

```diff
@@ -195,23 +197,34 @@ def upload_metadatas(
 
         uploaded_files = []
 
+        def get_file_digest(path: str, hashtype: str) -> str:
```

Comment:
We have a similar func in charon.utils.files, maybe we should re-use it by some modification.